training/src/main/scala/com/airbnb/aerosolve/training/SplineTrainer.scala

```diff
@@ -352,7 +510,7 @@ object SplineTrainer {
                   dropout : Double,
                   margin : Double) : Double = {
     val flatFeatures = Util.flattenFeatureWithDropout(fv, dropout)
-    val prediction = model.scoreFlatFeatures(flatFeatures)
+    val prediction = model.scoreFlatFeatures(flatFeatures) / (1.0 - dropout)
```

Comment:
I don't quite get this change. I thought when training with dropout, the updating process should be the same. But at testing time, we need to rescale based on the dropout rate: "At test time, we use the “mean network” that contains all of the hidden units but with their outgoing weights halved to compensate for the fact that twice as many of them are active." so we might need to make the change at the end of the whole training process and scale the weight by weight = weight * (1.0 - dropout rate)